RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+
+Each CC or CN has a local routing table that contains the online communication addresses of all CCs and CNs on the Tari communication network known to that CC or CN.
+When a CC or CN wants to join the Tari communication network, the online communication address of at least one other CN that is part of the network needs to be known.
+The online communication address of the initial CN can either be manually provided or a bootstrapped list of "reliable" and persistent CNs can be provided with the Validator Node, Base Node, Wallet or Asset Manager software.
+The new CC or CN can then request additional peer contact information of other CNs from the initial peers to extend their own routing table.
+
+The routing table consists of a list of peer addresses that link node ids, public identification keys and online communication addresses of each known CC and CN.
+
+The Peer Address stored in the routing table can be implemented as follows:
+
+| Description      | Data type          | Comments                                                                 |
+|---               |---                 |---                                                                       |
+| network address  | network_address    | The online communication address of the CC or CN                         |
+| node_id          | node_id            | Registration Assigned for VN, Self selected for BN, W and AM             |
+| public_key       | public_key         | The public key of the identification cryptographic key of the CC or CN   |
+| node_type        | node_type          | VN, BN, W or AM                                                          |
+| linked asset ids | list of asset ids  | Asset ids can be used as an address on Tari network similar to a node id |
+| last_connection  | timestamp          | Time of last successful connection with peer                             |
+
+When a new CC or CN wants to join the Tari communication network they need to submit a joining request to the rest of the network.
+The joining request contains the peer address of the new CC or CN.
+Each CN that receives the joining request can decide if they want to add the new CCs or CNs contact information to their local routing table.
+When a CN, that received the joining request, has a similar node id to the new CC or CN then that node must add the peer address to their routing table.
+All CNs with similar node ids to the new CC or CN should have a copy of the new peer address in their routing tables.
+
+To limit potential attacks, only one registration with the same online communication address can be stored in the routing table of a CN.
```

Comment:
- For IPv4 and IPv6 it would not be possible, but if Tor or I2P addresses were used I believe you could spin up more than one node behind the same VPN. Your bandwidth will be limited when doing this with I2P or Tor addresses. We can discuss this, but I think it might be a good thing to have this restriction.
- That is true, I also think that a user should be able to run a VN, BN, W and TW from one machine. I changed the wording of the sentence slightly to allow for this exception.